kernel/aster-nix/src/net/socket/mod.rs

```diff
@@ -63,18 +63,39 @@ pub trait Socket: FileLike + Send + Sync {
         return_errno_with_message!(Errno::EOPNOTSUPP, "setsockopt() is not supported");
     }
 
-    /// Receive a message from a socket
-    fn recvfrom(&self, buf: &mut [u8], flags: SendRecvFlags) -> Result<(usize, SocketAddr)> {
-        return_errno_with_message!(Errno::EOPNOTSUPP, "recvfrom() is not supported");
-    }
-
-    /// Send a message on a socket
+    /// Sends a message on a socket
     fn sendto(
```

Comment:
Just realized that maybe we should just move the `sendto`/`recvfrom` logic in this file to `syscall/sendto.rs` and `syscall/recvfrom.rs`?

As strong evidence, [the `struct proto` in the Linux kernel](https://elixir.bootlin.com/linux/v6.9.5/source/include/net/sock.h#L1200) has only `recvmsg` and `sendmsg` methods and _no_ `recvfrom` and `sendto` methods, which means that no protocol can override the default behavior of `recvfrom` and `sendto`. So it is safe for us to do the same.

Reply:
> Just realized that maybe we should just move the sendto/recvfrom logic in this file to syscall/sendto.rs and syscall/recvfrom.rs?

It makes sense to me. The problem is fixed.

> Also, FIXME: peer_addr should be `None` for recvmsg can be addressed. I don't think the Linux implementation writes the remote address in recvfrom for TCP sockets (maybe you can write a test case to verify this).

Yes, I verified that recvfrom does not writes remote address. The FIXME is removed.